code/Model/SiteTree.php

```diff
@@ -456,7 +456,7 @@ public static function get_by_link($link, $cache = true)
         $parentIDExpr = sprintf('"%s"."ParentID"', $tableName);
 
         $link = trim(Director::makeRelative($link) ?? '', '/');
-        if (!$link) {
+        if ($link === false || $link === null || $link === '') {
```

Comment:
Note that while `trim()` will only ever return a string, having the full check here avoids issues if we ever remove that trim.